lib/local_daily_digest.js

```diff
@@ -0,0 +1,448 @@
+var async = require('async'),
+	_ = require('lodash'),
+	log = require('../log'),
+	Profile = require('../models').Profile,
+	Contact = require('../models').Contact,
+	log = require('../log'),
+	mail = require('../mail');
+
+
+var	finalOutput = [];
+
+function get() {
+	var result = {},
+	 	allLocalProfilesWDigest = [],
+	 	tempCheckedOutUsers = [],
+		requiredLocationsId =[],
+	 	requiredUsers=[],
+	 	checkedOutUsers =[],
+		checkedInUsers = [];
+	
+	async.series([
+		function (cb) {
+			//Get a list all of profiles who have the daily digest array and are verified
+			Profile.find( {'dailyDigest': {$not: {$size: 0}}, 'verified': true }, function(err, profile){
+			if(err){
+		  		console.log("500", new Error(err));
+		  		return cb(true);
+		  	}
+
+		  	if(!profile){
+		  		console.log("400 No Profiles were found");
+		  	}
+
+		  	profile.forEach(function(profile){
+	  			allLocalProfilesWDigest.push(profile);
+		  	});	
+
+  			return cb();
+			});
+		},
+		function(cb){
+			//For all the profiles which want to get a digest, get a list of the locations they want a digest for
+			allLocalProfilesWDigest.forEach(function(profile){
+				profile.dailyDigest.forEach(function(location){
+					requiredLocationsId.push(location);
+				});
+			});
+			console.log("the countries these managers/editors want the dailyDigest for are: ", requiredLocationsId);
+			return cb();
+		},
+		function(cb){
+			//Get a list of all local profiles for those locations
+			async.forEachSeries(requiredLocationsId, function(location, callback2) {
+				Contact.find({ type: 'local', locationId: location}, function(err, contact){
```

Comment:
Now this is not being picky however :-) : this is still a huge query, which will return a LOT of contacts. Could you add directly to this query the condition that a contact should have checked in or out within the last 24 hours ? (much better to make mongo work rather storing all of these contacts in memory)